### DISCUSSION TITLE
Why/how does v3.0.1 resolve library dependencies that 2.5.2 didn't resolve?

### DISCUSSION THREAD

**Discussion by cdfarrow** [pythonnet/pythonnet#2211] [Q&A]
First of all, thank-you to the developers for this project, which I've been depending on for 15 years, and well done on the v3 update.

My project is a Python app that uses libraries from a large .NET application that has a host of dependency overrides in its .config file. With pythonnet 2.5.2, I've been including some old versions of the dll files to resolve the dependencies[^1], rather than using a .config file (this has been the path of least pain, since my Python app doesn't have a .exe file :-) )

Now, I'm trying to get it working with pythonnet 3.0.1, and strangely it is working without any of those local dlls. Many things in v3 are more strict (quite a few fixes have been needed), so it surprises me that something would be more lax. 

My question is: why is this now working in v3? Is it not enforcing version checking on library dependencies (i.e. effectively doing an automatic bindingRedirect)? If so, then this is really good news! However, I'm nervous about celebrating this new behaviour in case there's some hidden thing going on that might bite me with incorrect execution or elusive bugs, or something. Thanks for any insights!

[^1]: E.g. the main application uses and ships v2.9 of a dll, but some of the libraries that it uses depend on v2.8 of the library, so the main application's .config file has a bindingRedirect to accept v2.9 in all cases. In my app I ship v2.8 and do a clr.AddReference() to it.

**Comment by pdetheridge** [comment]
I would also be really interested in this answer, as I can also see moving from 2.5.2 to 3.0.1 resolves dll load issues I was having with a mixture of versions of dependencies.  I can see post AddReference it is bringing in more of the dependent dll's and everything seems to be working, so is it in effect doing an automatic bindingRedirect?  Which would be great, just as above would be good to confirm (ie. there is nothing else to watch out for).
